### PR TITLE
GAWB-139: Fix rendering when stuff is a vector, not a string

### PR DESCRIPTION
**Pull request by MattPutnam**
- Added safety if trying to parse a GCS URL and the input is not a string
- Several fixes for rendering of attributes in data tables
- Fix for rendering fields in the workflow details
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] **Submitter**: Tell tech lead that the PR exists if s/he wants to look at it
- [x] **Submitter**: Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x] **Tech lead**: sign off
- [x] **LR**: sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't 
  working for ANY reason!
- [x] **Submitter**: Verify swagger UI on dev server still works after deployment
- [x] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/org/broadinstitute/firecloud_ui/common.cljs [hunk 1]
```diff
@@ -104,15 +104,17 @@
   (str (config/api-url-root) "/cookie-authed/download/b/" bucket "/o/" object))
 
 (defn gcs-uri->download-url [gcs-uri]
-  (let [matcher (re-find #"gs://([^/]+)/(.+)" gcs-uri)]
-    (when (= 3 (count matcher)) ;; first match will be the whole thing
-      (gcs-object->download-url (matcher 1) (matcher 2)))))
+  (when (string? gcs-uri)
+    (let [matcher (re-find #"gs://([^/]+)/(.+)" gcs-uri)]
+      (when (= 3 (count matcher)) ;; first match will be the whole thing
+        (gcs-object->download-url (matcher 1) (matcher 2))))))
```
Comment (davidangb) [inline src/cljs/org/broadinstitute/firecloud_ui/common.cljs]: I know this wasn't your doing originally ... but could gcs-uri->download-url internally call parse-gcs-uri, so we don't have the duplicate regex code?

Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/common.cljs]: Good idea.

**Comment by davidangb** [comment]
:+1: at the code level. Do you have a workspace with data/submissions/whatever I can test this against?

**Comment by davidangb** [comment]
I found the array_inputs2 workspace that has array inputs for a submission. Do you have a workspace that tests the gcs uri code?

**Comment by MattPutnam** [comment]
Looks like `gcs-uri->download-url` isn't even used anywhere.

**Comment by davidangb** [comment]
1) if it's not used, could you please remove it? I realize this is scope-creeping.

2) then, do you have a workspace that tests the parse-gcs-uri changes? In other words, you made some changes to rendering of attribute values related to gcs uris; where can I test that?

**Comment by MattPutnam** [comment]
In that array_inputs2 workspace, look at any of the workflows in the monitor tab.  Previously this code was crashing because the inputs/outputs were vectors, and re-find was breaking since it asserts that it's given a string.

Some of those workflow values are broken, but that's because the method config was set up wrong.  The UI is correctly displaying the broken values though.

**Comment by davidangb** [comment]
(we talked outside of this PR) - okay, calling this done :+1: . Feel free to merge.

**Pull request merged**